### DISCUSSION TITLE
How to change input method in mpv on linux?

### DISCUSSION THREAD

**Discussion by Tony15246** [mpv-player/mpv#14775] [Q&A]
I found out that whatever input method I use, I can only type English words in mpv console. My input method framework is fcitx5 and my platform is linux, I'm using KDE Plasma and my window manager is Wayland.

I'm writing a plugin to search for ass files online and I need to input Chinese words to search, but I can not switch to Chinese input method. I've searched online for input method related problem in mpv. And I see someone mentioned that mpv has IME support on Windows. I tried with my windows computer and I can switch to Chinese input method and type Chinese words in mpv.

So I'd like to know that does mpv have IME support on linux? If mpv does support IME on linux, then how should I config mpv to activate it? Or does my problem has nothing to do with mpv and it's just the problem of my input method framework config?

I'm using arch linux so I download mpv from [this official package](https://archlinux.org/packages/extra/x86_64/mpv/). And the mpv version is 0.38.0.

As for input method engine, I'm using Rime Input Method Engine from [fcitx5-rime](https://archlinux.org/packages/extra/x86_64/fcitx5-rime/) and [rime-ice-git](https://github.com/iDvel/rime-ice) as config patch.

**Comment by guidocella** [comment]
mpv supports IME on Xorg. To use it Wayland it still needs to implement the input method protocol, but I don't know if protocols that are not yet standardized are accepted.

**Reply by Tony15246** [reply]
Thank you very much for your answer! I switched to X11 session and tried again. However, the input method still does not work. Maybe the problem is not solely cause by window manager. Is there probably any specific settings I need to set to the config of mpv?

**Reply by guidocella** [reply]
Then I guess IME doesn't work on Xorg either and was only implemented on Windows.

**Reply by Tony15246** [reply]
I suddenly discovered that my clipboard worked with X11 session and I can paste text into mpv, which never happened in Wayland session, and, to some extent, sloved my problem. Though not a perfect solution, at least I'm now able to paste any text whether English or Chinese into mpv. Anyway, thank you again for your help and wish you have a good day.😊

**Reply by guidocella** [reply]
Yeah we only supporting pasting on wlroots-based compositors with `wl-copy`. Eventually we will add proper clipboard support (https://github.com/mpv-player/mpv/pull/13837).

**Reply by guidocella** [reply]
Actually it should work on KDE Wayland if you put a script named `wl-paste` in `PATH` that calls `qdbus org.kde.klipper /klipper getClipboardContents`.

**Reply by Tony15246** [reply]
I'm gonna tried it out later. Thank for your tips. Actually, I thinked of pasting text into mpv right after I had found out that I can not use my input method. But the official doc of mpv only gives out the keyboard shoutcut of pasting and has no explaination of how clipboard works. I searched online but found nothing useful, only tons of tutorial about how to copy things like youtube url into mpv to play videos. The internet is overflowed with information nowadays and so hard to search for what you want.😂
